modules/builds-docker-credentials-private-registries.adoc

```diff
@@ -87,6 +79,14 @@ the build configuration:
 $ oc set build-secret --push bc/sample-build dockerhub
 ----
 
+. You can also link the push secret to the *builder* service account instead of specifying the
+`pushSecret` field. Push secrets linked to the *builder* service account can be automatically added
+to any build in the namespace.
```

Comment:
@bparees @bmcelvee Moved the secret linking as an optional step in the procedure. Please let me know if this phrasing makes sense.

Reply:
it doesn't.

push secrets don't have to be linked at all.  we only search the build's SA for linked secrets for pulling.  We don't do any logic to search for push secrets, either we use what we found for pull, or we use the pushsecret you explicitly specified (which, again, does not need to be linked to the SA anymore)

Reply:
> We don't do any logic to search for push secrets

I should qualify this.....  as far as i know we don't.  my memory on this isn't great, so i could be wrong, but i don't think we do.

Reply:
The relevant function is `resolveImageSecretAsReference`. We call it in the following locations:

1. Resolve the push secret: https://github.com/openshift/openshift-controller-manager/blob/master/pkg/build/controller/build/build_controller.go#L1113-L1122
2. Resolve the pull secret: https://github.com/openshift/openshift-controller-manager/blob/master/pkg/build/controller/build/build_controller.go#L1144-L1162
3. Resolve the pull secret for image sources: https://github.com/openshift/openshift-controller-manager/blob/master/pkg/build/controller/build/build_controller.go#L1165-L1176

So it looks like I should also add similar text for the image sources.

Reply:
ok, i stand corrected..in that case my comments from the other section apply here as well in terms of better describing the actual flow that happens to pick a secret